control-plane-operator/controllers/hostedcontrolplane/etcd/status.go

```diff
@@ -0,0 +1,170 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ReconcileControlPlaneComponent creates/updates the ControlPlaneComponent resource for etcd
+// This is needed for other v2 components that depend on etcd
+func ReconcileControlPlaneComponent(
+	ctx context.Context,
+	hcp *hyperv1.HostedControlPlane,
+	shards []hyperv1.ManagedEtcdShardSpec,
+	client client.Client,
+	version string,
+) error {
+	cpc := &hyperv1.ControlPlaneComponent{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "etcd",
+			Namespace: hcp.Namespace,
+		},
+	}
+
+	// Get aggregate status
+	condition, err := AggregateShardStatus(ctx, hcp, shards, client)
+	if err != nil {
+		return err
+	}
+
+	// Build resource list from all shards
+	var resources []hyperv1.ComponentResource
+	for _, shard := range shards {
+		stsName := manifests.EtcdStatefulSetForShard(hcp.Namespace, shard.Name).Name
+		resources = append(resources, hyperv1.ComponentResource{
+			Group: "apps",
+			Kind:  "StatefulSet",
+			Name:  stsName,
+		})
+	}
+
+	// Check if ControlPlaneComponent exists
+	err = client.Get(ctx, types.NamespacedName{Namespace: hcp.Namespace, Name: "etcd"}, cpc)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		// Create new ControlPlaneComponent
+		cpc.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: hyperv1.GroupVersion.String(),
+				Kind:       "HostedControlPlane",
+				Name:       hcp.Name,
+				UID:        hcp.UID,
+			},
+		}
+		if err := client.Create(ctx, cpc); err != nil {
+			return err
+		}
+		// Need to Get it again to update status
+		if err := client.Get(ctx, types.NamespacedName{Namespace: hcp.Namespace, Name: "etcd"}, cpc); err != nil {
+			return err
+		}
+	}
+
+	// Update status (works for both newly created and existing resources)
+	now := metav1.Now()
```

Comment:
You can't assume last transition time is always now, it should only change when the status flicks true to false on the condition. If you keep reconciling this status you'll just keep ticking the time up for no reason